target_chains/ethereum/sdk/solidity/PriceCombine.sol

```diff
@@ -0,0 +1,167 @@
+// PriceCombine.sol
+// SPDX-License-Identifier: Apache 2
+pragma solidity ^0.8.0;
+
+import "./PythStructs.sol";
+
+// Constants for working with Pyth's number representation
+int32 constant PD_EXPO = -9;
+uint64 constant PD_SCALE = 1_000_000_000;
+uint64 constant MAX_PD_V_U64 = (1 << 28) - 1;
+
+// Solidity Library to easily combine 2 Pyth Prices into 1 ( for example SOL:USD/ETH:USD=SOL/ETH). For Rust implementation, refer Pyth Rust SDK https://github.com/pyth-network/pyth-sdk-rs/blob/main/pyth-sdk/src/price.rs
+// Based on the Rust implementation, the functions are defined within the `impl Price` block, and they are public methods of the `Price` struct and are as follows `get_price_in_quote`, `div`, `normalize`, `scale_to_exponent`
+// Similarly the functions implemented in solidity are `public` and `pure` types.
+
+library PriceCombine {
+    /**
+     * @notice Computes the price in quote currency.
+     * @param self The price of the base currency.
+     * @param quote The price of the quote currency.
+     * @param resultExpo The exponent for the result.
+     * @return The combined price in the quote currency.
+     */
+    function getPriceInQuote(PythStructs.Price memory self, PythStructs.Price memory quote, int32 resultExpo) public pure returns (PythStructs.Price memory) {
+        PythStructs.Price memory result = div(self, quote);
+        // Return zero price so the error can be gracefully handled by the caller
+        if (result.price == 0 && result.conf == 0) {
+            return PythStructs.Price({
```

Comment:
why not put this check in `scaleToExponent`?

Reply:
I will try to update the check in `scaleToExponent`